### DISCUSSION TITLE
Login & Registration

### DISCUSSION THREAD

**Discussion by jessejamesrich** [npm/feedback#731] [Website]
I dunno how to say this as a dev in the field for 20 or so years without sounding like a dink, but the login/registration piece of npmjs.com is potentially the worst I've come across in a long time. There was a short period where Binance had a time-limited multi-factor-authentication (4 factors!!) that was one of the worst experiences in the history of the net. It's not that bad, but npmjs.com is pretty bad for 2022. I dunno how. Firstly, I'm trying to build and ship my first package, so I know this is going to have hiccups, and also I'm only about 1 - 1.5 years into node/Reach. Previous to that 15-year lamp, 5 on ASP/MS. I feel like there's gonna be hiccups and they're gonna be my fault, but at the current moment, I can even reset my password which is not a "me" thing. 

1. Get here, sign up with my personal account. Not sure why, probably because sometimes I do things for me and not my CO. all goes pretty well. MFA sets up pretty well and I can log in or logout without issue. 

2. This is not actually what I need though, I need an organization because. I've got a boatload of proprietary packages. So I'm given the option to convert, but this reads as though my existing private account megaballs30069 will become my corporate account and I'll end up with a new account I may have misread this to be fair, but now I'm kinda stuck because most times getting this wrong hoops you. There's a similar process on github that helps one move from individual to organization painlessly. This was not helpful in the least.

3. What's the safest thing? Setup up a separate corporate account so I cab be billed correctly and build a team around an or. And again I get this done with few issues. Github makes this so simple; npmjs.org - at this point, it's not terrible. It's just not great,. Once there I send out some invites to myself and others. No reflection of invites I could find in the system to the invitees, so huge ux problem there - how do they know to join? But the email was sent which informed me - great. I guess that's half good. Oddly, you can invite someone and they simply won't show up on your list of invitees. So you add one email address over and over again and the alter window (which is at the top and above the visible window width) is sending out warnings you can see. So far, this is all really gross. 

4. I decide to start things over simpler again. I've not got three accounts in npmjs. Two corporate and one personal. I decide let's did one corporate (jesse@) because the username doesn't match my common GitHub-like username. So I ditch the corporate, personal email and the personal account. I shut down both accounts. Now I just have one account. It's an organization account. It really should be the one ring to rule them all. Incidentally, session management is being done painlessly as chrome has each email tied to a browser instance and those accounts run on google workspace/domains. So this should be easy. npmjs should have seen 3 separate accounts. Not it should see one.

5. I can start over again by adding a corporate personal (name@compain.domain) account with a standard username I use in respect to my and my company. Should be easy right? I just deleted the two accounts in conflict. So I'll go through the capture yet again and oh shit of course I can actually register a username that I registered and deleted in the last 5 minutes. I get it, I think, but this not how Girhub works. Or Slack channels. Changing things changes them. Deleting deleted. 

6. Do the captcha again now or the 40th time. I'm half checked out, in another tab I'm looking up alternatives to npmjs.com. Like, this is really just trash usability. Captcha tells you to identify something, but only on the first screen of course - if you switch tabs, someone walks in to chat, shit? What I looking for again? Busses, boats, or bikes? Start over. Sigh. Start over. 

7. So here we are. Having three accounts to try and understand this new service I've never used before. 2/3rds of which are inaccessible or dead already. I appreciate the notion of usernames, deleting them, the knock-on effect, and getting them right. This mostly all stems from moving to an organization and not knowing how it would be repped online. As PunkBallSalami45839 or it's company name. I also get that usernames are tied to repos and willy-nilly deleting them is a real problem. But for me, this is a way bigger problem in Github and I've had zero problems in my life understanding how to get through their stuff.

8. Here's the kicker - at some point my corporate account which also has a wrong username gets logged out. I guess it's good it has a wrong username because if it's gonna be the org account, I can delete it and cancel my membership and start fresh. At least I'll have 1 of 3 npsjs accounts working. Woo. Almost a weird blessing, except... yeah, so now I can't access it. I can no longer login with that account. Perhaps because the members have been deleted? The username and/or email no longer work. I dunno why. Chrome shows the password as correct, but can't login. 

9. Go to reset password: Easy enough, get the email super quick, which is great and the link works, which is great, but why is there all this extra and confusing garbage in here. We've been resetting passwords for 30 years and for some reason we have to do it differently here:

First screen: 

```
Reset password
If you have forgotten your password, click continue.

This will generate a new, temporary password for your account. After you have used this temporary password to log in, you can create your own password.
```

Makes perfect sense. 

```
Password Reset
Your temporary password for junkface1000 is:

owtoNbXqiyWDzlD6wABm0ja3vR4=

Please copy the password and login, after which you will be asked to set a new password.
```

This is dumb. Just reset the password right here. Further, the button that says log in looks like it's disabled due to the color. So, very confusing. But, whatever, got my temp password and...... ... ... .. .. ..... . .. . . . ...... 

It doesn't work. First I get bad username and password, then I get a 404 eventually. Basically all 3 accounts are useless. You'll also note I went with junkface1000 above: stupid of me as it's a username. And the usernames can be changed nor used again, but none of that really matters since logging in and creating an organization is crazy - 

9 times in 10 I'd say it was a buggy release. 5 times in 10 I'd say I'm an idiot for just not giving proper consideration to how I wanted to structure this all. But gotta realize, this is a brand new SaaS to me that is based off other SaaS workflows that make sense and work. I'm in a testing mode to see if Im getting what I need. So I'll take some heat, but hands down this is bad work. This is Mailchimp UI bad work IMHO. Which is absolutely crazy to me if it's a github product subsidiary, because that's one of the best products ever. 

I dunno. This isn't mean to attack anyone. This is just a dude who does this for a living saying this is a complete s show.

**Comment by MylesBorins** [comment]
@jessejamesrich it is coming across loud and clear that you are frustrated, but you are going to need to dial it back a bit if you want to have a productive conversation here, in particular your decision to use human anatomy in the example user names isn't very helpful nor is the direct attacks of our site as well of the work of others.

It is really hard to follow your overall narrative here, which seems to be critiquing a number of different aspects of the npm experience ranging from:
* account creation
* accounts vs organizations
* organization invites
* password reset

One thing that it seems like needs to be clarified, there are no such thing as corporate or organization accounts. Organizations can be used to manage groups of packages + maintainers, they are named and owned the scope with the same name as the organization. Accounts also have the ability to publish packages to the scope with the same name as the account. An account can be converted to an organization, this is a permanent change (although or support team can roll this back if you open a ticket). An organization requires an owner.

If you have personally created accounts / organizations and lost access you can open a support ticket to help you.

For next steps here. To turn your experience into actionable changes it would be helpful to separate the specific areas of the experience you had issues with and clearly identify what the problem was. e.g. "password reset giving you a temporary password to then enter in to login only to have you pick another password is painful". I happen to agree with this and a review / revamp of our overall password reset process is currently in our backlog.

If you could make a thread for each specific experience concern it will be much easier for me to collaborate and think about solutions.

**Reply by jessejamesrich** [reply]
I will do that, and my sincerest apologies. Honestly, I am mortified to go back and look based on what you have here. It is uncharacteristic of me to fly in hot, even at the worst of times. I will get this sorted out and fixed up with proper and appropriate apologies where needed and get an entire list of different issues. I will do better. Will follow up.

**Reply by jessejamesrich** [reply]
Ok. Took a while to get back here, reminded by the reply that says simply good. Gotta be honest, not feeling I put a lot of mustard into that post. I can't tell you why I indeed used a juvenile pacesetter username - it's very bizarre behaviour on my part at any time but especially during a time in which I'm trying to get a corporate site setup. As this seemingly was me, it then seems self-evident I'm in no way levying a charge at y'all you issue usernames unless you do issue usernames?

One thing I'll give ya is that my post is hard to follow. Really hard to follow. Aimless bitch sessions with no actionable items aren't generally things I aim to fill my midnights with, but even in re-reading the thing a few times, I've taken the time to use specificity as I was like taking notes in tandem with simply getting account working. So, though this OP should be shot out of a cannon into a lake of liquid hot molten lava so we can all move on. The takeaway for me some two months later upon return is that support team required more specificity to compose an actionable plan for fixes while the sheer magnitude of unexpected behaviour made in exceptionally dificult to do.

This OP is the crash location. It's not the black box.

This thread should be entirely burned with fire, and I come up with a more concise way of communicating the failures (if they still exist). Instead, I'm included to believe two months later, whatever happened those days has resolved. In retrospect, I'm having a hard time understanding how NPM could have such a glaring set of problems, while someone with my experience couldn't pull off basic things.

This whole goddamned thing was a ghost in a machine, I bet.

**Comment by khamidj0n** [comment]
good
